src/main/scala/com/sourcegraph/sbtsourcegraph/SourcegraphEnable.scala

```diff
@@ -143,4 +164,34 @@ object SourcegraphEnable {
     semanticdbVersion = Versions
       .semanticdbVersion(overriddenScalaVersion.getOrElse(projectScalaVersion))
   } yield (p, semanticdbVersion, overriddenScalaVersion)
+
+  def javacModuleOptions: List[String] =
+    List(
+      "-J--add-exports",
+      "-Jjdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+      "-J--add-exports",
+      "-Jjdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
+      "-J--add-exports",
+      "-Jjdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED",
+      "-J--add-exports",
+      "-Jjdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
+      "-J--add-exports",
+      "-Jjdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED"
+    )
+
+  private def calculateJavaHome = {
+    // We can safely use java.home property
+    // on JDK 17+ as it won't be pointing to JRE which
+    // doesn't contain a compiler.
+    if (Versions.isJavaAtLeast(17)) {
+      // On JDK 17+ we need to explicitly fork the compiler
+      // so that we can set the necessary JVM options to access
+      // jdk.compiler module
+      Some(new File(System.getProperty("java.home")))
```

Comment:
I'm not 100% sure, but forking javac may cause some builds to fail. Not a blocking comment, just good to keep an eye for it. If we can't turn on forking then we could try to automatically add the exports to SBT_OPTS

Reply:
Can you expand on what might be causing issues?

Perhaps I could add an extra test for this just to be sure

Reply:
I can't remember any concrete examples. Fine to try this out and just keep an eye out for regressions.